rocAL/rocAL_pybind/amd/rocal/plugin/pytorch.py

```diff
@@ -276,44 +204,45 @@ class ROCALClassificationIterator(ROCALGenericIterator):
     fill_last_batch = True, last_batch_padded = True   -> last batch = [7, 7], next iteration will return [1, 2]
     fill_last_batch = True, last_batch_padded = False  -> last batch = [7, 1], next iteration will return [2, 3]
     """
+
     def __init__(self,
                  pipelines,
                  size = 0,
-                 auto_reset=False,
-                 fill_last_batch=True,
-                 dynamic_shape=False,
-                 last_batch_padded=False,
-                 display=False,
-                 device="cpu",
-                 device_id =0):
+                 auto_reset = False,
+                 fill_last_batch = True,
+                 dynamic_shape = False,
+                 last_batch_padded = False,
+                 display = False,
+                 device = "cpu",
+                 device_id = 0,):
         pipe = pipelines
         super(ROCALClassificationIterator, self).__init__(pipe, tensor_layout = pipe._tensor_layout, tensor_dtype = pipe._tensor_dtype,
-                                                            multiplier=pipe._multiplier, offset=pipe._offset,display=display, device=device, device_id = device_id)
-
+                                                          multiplier = pipe._multiplier, offset = pipe._offset, dsiplay = display, device = device, device_id = device_id)
 
-class ROCAL_iterator(ROCALGenericImageIterator):
-    """
-    ROCAL iterator for classification tasks for PyTorch. It returns 2 outputs
-    (data and label) in the form of PyTorch's Tensor.
-
-    """
-    def __init__(self,
-                 pipelines,
-                 size = 0,
-                 auto_reset=False,
-                 fill_last_batch=True,
-                 dynamic_shape=False,
-                 last_batch_padded=False):
-        pipe = pipelines
-        super(ROCAL_iterator, self).__init__(pipe)
 
 
-def draw_patches(img,idx, bboxes):
-    #image is expected as a tensor, bboxes as numpy
+def draw_patches(img,idx,bboxes):
+    #image is expected as a tensor
     import cv2
-    img=img.cpu()
+    img = simg.cpu()
     image = img.detach().numpy()
     image = image.transpose([1,2,0])
     image = cv2.cvtColor(image, cv2.COLOR_RGB2BGR )
     image = cv2.UMat(image).get()
     cv2.imwrite(str(idx)+"_"+"train"+".png", image)
+    try:
+        path = "OUTPUT_IMAGES_PYTHON/NEW_API/PYTORCH/"
+        isExist = os.path.exists(path)
+        if not isExist:
+            os.makedirs(path)
+    except OSError as error:
+        print(error)
+    for (l, t, r, b) in bboxes:
```

Comment:
is there a case where bboxes is empty which is not handled here

Reply:
Handled that case too